src/parser/core/tests/CombatLogParserIntegrationTest.js

```diff
@@ -0,0 +1,117 @@
+import { i18n } from 'interface/RootLocalizationProvider';
+import ParseResults from 'parser/core/ParseResults';
+import BaseChecklist from 'parser/shared/modules/features/Checklist2/Module';
+import renderer from 'react-test-renderer';
+import { loadLog, suppressLogging, parseLog } from './log-tools';
+import { statistic, expectSnapshot } from './snapshotTest';
+
+function integrationStatistic(analyzer, parser) {
+  if(!analyzer.active) {
+    return 'module inactive';
+  }
+  if(!analyzer.statistic || analyzer.statistic() === undefined) {
+    return 'module has no statistic method';
+  }
+  return statistic(analyzer, parser);
+}
+
+function integrationSuggestions(analyzer) {
+  if(!analyzer.active) {
+    return 'module inactive';
+  }
+  if(!analyzer.suggestions) {
+    return 'module has no suggestions';
+  }
+  const results = new ParseResults();
+  analyzer.suggestions(results.suggestions.when, { i18n });
+  return results.issues;
+}
+
+function checklist(parser) {
+  const checklistModule = Object.values(parser.constructor.specModules).find(m => m.prototype instanceof BaseChecklist);
+  if(checklistModule === undefined) {
+    return 'no checklist';
+  }
+  const result = parser.getModule(checklistModule).render();
+  return renderer.create(result).toJSON();
+}
+
+/**
+ * Generates an integration test for a spec's CombatLogParser instance.
+ *
+ * The integration test applies the parser to a single fight that is
+ * downloaded and stored in the repository.
+ *
+ * `console.log` and `console.warn` are suppressed during the test, but
+ * `console.error` is not. This behavior can be disabled with the
+ * `suppressWarn` and `supressLog` parameters.
+ *
+ * To obtain the necessary JSON, load the report of interest and get the
+ * URLs for each call to the WoWAnalyzer API from the 'Network' section
+ * of the developer tools. Download the contents of each URL and store
+ * them somewhere. The full set of events should be gzipped (which
+ * reduces space usage from several MB to ~100kb).
+ *
+ * See the Brewmaster analyzer for a worked example.
+ *
+ * @param {object} parserClass - (uninstantiated) CombatLogParser subclass to test.
+ * @param {string} key - Key identifying which log from `test-logs` to load
+ * @param {boolean} suppressWarn - Suppress `console.warn`
+ * @param {boolean} suppressLog - Suppress `console.log`
+ */
+export default function integrationTest(parserClass, key, suppressWarn=true, suppressLog=true) {
+  return () => {
+    let log;
+    beforeAll(() => {
+      return loadLog(key).then(res => { log = res; });
+    });
+
+    suppressLogging(suppressLog, suppressWarn, false);
+
+    it('should parse the example report without crashing', () => {
+      const parser = parseLog(parserClass, log);
```

Comment:
Why isn’t this in the beforeAll?

Reply:
honestly, this should just go away.

initially it was because we want the error to be associated with the test and not the `beforeAll`, but since tests happen asynchronously this may not actually run before other tests that use the `beforeAll` parser.